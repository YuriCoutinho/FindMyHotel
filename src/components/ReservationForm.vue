<template>
  <form @submit.prevent="submitForm" class="space-y-6">
    <div class="form-group">
      <label for="firstName" class="block text-sm font-medium text-gray-700">Nome:</label>
      <InputText
        id="firstName"
        v-model="firstName"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <div class="form-group">
      <label for="lastName" class="block text-sm font-medium text-gray-700">Sobrenome:</label>
      <InputText
        id="lastName"
        v-model="lastName"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <div class="form-group">
      <label for="email" class="block text-sm font-medium text-gray-700">Email:</label>
      <InputText
        id="email"
        v-model="email"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <div class="form-group">
      <label for="number" class="block text-sm font-medium text-gray-700">Número do celular:</label>
      <InputMask
        id="number"
        v-model="phoneNumber"
        mask="(99) 99999-9999"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
      />
    </div>
    <div class="form-group">
      <label for="documentNumber" class="block text-sm font-medium text-gray-700">CPF:</label>
      <InputMask
        id="documentNumber"
        v-model="documentNumber"
        mask="999.999.999-99"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <div class="form-group">
      <label for="payment" class="block text-sm font-medium text-gray-700"
        >Cartão de Crédito:</label
      >
      <InputMask
        id="payment"
        v-model="payment"
        mask="9999 9999 9999 9999"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <div class="form-group">
      <label for="cvv" class="block text-sm font-medium text-gray-700">CVV:</label>
      <InputMask
        id="cvv"
        v-model="cvv"
        mask="999"
        class="w-full px-3 py-2 border border-gray-300 rounded-md"
        required
      />
    </div>
    <Button
      label="Reservar"
      type="submit"
      class="w-full text-white font-medium py-2 rounded-md bg-emerald-500 hover:bg-emerald-700 active:bg-emerald-900"
    />
  </form>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import InputText from 'primevue/inputtext'
import InputMask from 'primevue/inputmask'
import Button from 'primevue/button'

const firstName = ref('')
const lastName = ref('')
const email = ref('')
const phoneNumber = ref('')
const documentNumber = ref('')
const payment = ref('')
const cvv = ref('')

const emits = defineEmits(['reserve'])

const submitForm = () => {
  const reservationDetails = {
    firstName: firstName.value,
    lastName: lastName.value,
    email: email.value,
    phoneNumber: phoneNumber.value,
    documentNumber: documentNumber.value,
    payment: payment.value,
    cvv: cvv.value
  }

  emits('reserve', reservationDetails)
}
</script>
