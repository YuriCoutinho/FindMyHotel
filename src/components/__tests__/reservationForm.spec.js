import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import ReservationForm from '@/components/ReservationForm.vue'
import PrimeVue from 'primevue/config'
import InputText from 'primevue/inputtext'
import InputMask from 'primevue/inputmask'
import Button from 'primevue/button'

describe('ReservationForm.vue', () => {
  const createWrapper = () => {
    return mount(ReservationForm, {
      global: {
        plugins: [PrimeVue],
        components: { InputText, InputMask, Button }
      }
    })
  }

  it('Should render the component', () => {
    const wrapper = createWrapper()
    expect(wrapper.exists()).toBe(true)
  })

  it('Should initialize form fields correctly', () => {
    const wrapper = createWrapper()
    expect(wrapper.vm.firstName).toBe('')
    expect(wrapper.vm.lastName).toBe('')
    expect(wrapper.vm.email).toBe('')
    expect(wrapper.vm.phoneNumber).toBe('')
    expect(wrapper.vm.documentNumber).toBe('')
    expect(wrapper.vm.payment).toBe('')
    expect(wrapper.vm.cvv).toBe('')
  })

  it('Should match the snapshot', () => {
    const wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
