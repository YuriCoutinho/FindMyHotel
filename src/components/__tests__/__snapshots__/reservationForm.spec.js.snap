// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReservationForm.vue > Should match the snapshot 1`] = `
"<form class="space-y-6">
  <div class="form-group"><label for="firstName" class="block text-sm font-medium text-gray-700">Nome:</label><input type="text" class="p-inputtext p-component w-full px-3 py-2 border border-gray-300 rounded-md" id="firstName" required="" data-pc-name="inputtext" data-pc-section="root" value="" pc9=""></div>
  <div class="form-group"><label for="lastName" class="block text-sm font-medium text-gray-700">Sobrenome:</label><input type="text" class="p-inputtext p-component w-full px-3 py-2 border border-gray-300 rounded-md" id="lastName" required="" data-pc-name="inputtext" data-pc-section="root" value="" pc10=""></div>
  <div class="form-group"><label for="email" class="block text-sm font-medium text-gray-700">Email:</label><input type="text" class="p-inputtext p-component w-full px-3 py-2 border border-gray-300 rounded-md" id="email" required="" data-pc-name="inputtext" data-pc-section="root" value="" pc11=""></div>
  <div class="form-group"><label for="number" class="block text-sm font-medium text-gray-700">Número do celular:</label><input type="text" class="p-inputtext p-component p-inputmask w-full px-3 py-2 border border-gray-300 rounded-md" data-pc-name="pcinputtext" data-pc-extend="inputtext" data-pc-section="root" value=""></div>
  <div class="form-group"><label for="documentNumber" class="block text-sm font-medium text-gray-700">CPF:</label><input type="text" class="p-inputtext p-component p-inputmask w-full px-3 py-2 border border-gray-300 rounded-md" data-pc-name="pcinputtext" data-pc-extend="inputtext" data-pc-section="root" value=""></div>
  <div class="form-group"><label for="payment" class="block text-sm font-medium text-gray-700">Cartão de Crédito:</label><input type="text" class="p-inputtext p-component p-inputmask w-full px-3 py-2 border border-gray-300 rounded-md" data-pc-name="pcinputtext" data-pc-extend="inputtext" data-pc-section="root" value=""></div>
  <div class="form-group"><label for="cvv" class="block text-sm font-medium text-gray-700">CVV:</label><input type="text" class="p-inputtext p-component p-inputmask w-full px-3 py-2 border border-gray-300 rounded-md" data-pc-name="pcinputtext" data-pc-extend="inputtext" data-pc-section="root" value=""></div><button class="p-button p-component w-full text-white font-medium py-2 rounded-md bg-emerald-500 hover:bg-emerald-700 active:bg-emerald-900" type="submit" aria-label="Reservar" data-pc-name="button" data-p-disabled="false" data-pc-section="root" pc12="">
    <!----><span class="p-button-label" data-pc-section="label">Reservar</span>
    <!---->
  </button>
</form>"
`;
